app/controllers/home_controller.rb

```diff
@@ -21,4 +21,33 @@ def post_comments
 			redirect_to root_url
     end
   end
+
+  def like_post_and_comment
+    @like = Like.find_by_likeable_id_and_likeable_type_and_user_id(params[:likeable_id], params[:likeable_type], current_user.id)
```

Comment:
This could be 
```ruby
Like.find_by_likeable_id_and_likeable_type_and_user_id(like_params.merge({ user_id: current_user.id }))
```

with that you won't need this method(`find_by_likeable_id_and_likeable_type_and_user_id`) you can use `find_by` with that hash

Reply:
There is a simpler way also:
```ruby
@like = Like.where(like_params.merge({ user_id: current_user.id }).first_or_initialize
```
and then there is a method in rails to check if the record is new [new_record?](https://apidock.com/rails/ActiveRecord/Persistence/new_record%3F) and [persisted?](https://apidock.com/rails/ActiveRecord/Persistence/persisted%3F) which are opposite to each other so lets use them:

```ruby
if (@like.new_record? && @like.save) || (@like.persisted? && @like.save)
  @message = 'success'
  format.js
else
  @message = 'Sorry, something went wrong'
  format.js
end
```